scripts/makefile_app_params_check.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict
+from collections import defaultdict
+import json
+
+
+APPLICATION_FLAG_BOLOS_SETTINGS = int("0x200", 16)
+
+APP_LOAD_PARAMS_ALLOWED = [
+    "targetId",
+    "targetVersion",
+    "apiLevel",
+    "fileName",
+    "icon",
+    "curve",
+    "path",
+    "path_slip21",
+    "appName",
+    # "signature", # Reserved for internal usage
+    # "signApp", # Reserved for internal usage
+    "appFlags",
+    # "bootAddr", # Deprecated?
+    # "rootPrivateKey", # Should not be used for app deployment
+    # "signPrivateKey", # Should not be used for app deployment
+    # "apdu", # Should not be used for app deployment
+    # "deployLegacy", # Deprecated?
+    "delete",
+    # "params", # Deprecated?
+    "tlv",
+    "dataSize",
+    "appVersion",
+    # "offline", # Should not be used for app deployment
+    # "offlineText", # Should not be used for app deployment
+    # "installparamsSize", # Deprecated?
+    "tlvraw",
+    # "dep", # Deprecated?
+    "nocrc",
+]
+
+APP_LOAD_PARAMS_VALUE_CHECK = [
+    "curve",
+    "path",
+    "path_slip21",
+    "appName",
+    "appFlags"
+]
+
+
+def parse_listapploadparams(app_load_params_str: str) -> Dict:
+    # Convert to dict. Store value in list type as some params can appear
+    # multiple times (e.g `--path`).
+
+    app_load_params = defaultdict(list)
+    for param in app_load_params_str.split("--"):
+        param = param.strip()
+        if not param:
+            continue
+
+        if param.startswith("targetVersion="):
+            parts = param.split("=")
+        else:
+            parts = param.split(" ")
+
+        param_name = parts[0]
+
+        param_value = None
+        if len(parts) > 1:
+            param_value = " ".join(parts[1:])
+
+        app_load_params[param_name].append(param_value)
```

Comment:
About parameter parsing problems, could we delegate that painful task to python's `argparse` directly? I think using this PR https://github.com/LedgerHQ/ledger-secure-sdk/pull/161 to get a dictionary (from a TOML file) could be an easy way to do so in a standardized way, since the flags we're after here are made to be passed to a python script that uses argparse anyway (ledgerblue.loadApp)